nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -2202,11 +2576,50 @@ fn find_matching_rw_regions_in_volume(
     Ok(())
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    use crate::db::datastore::test_utils::datastore_test;
+impl DataStore {
+    pub async fn find_volumes_referencing_socket_addr(
+        &self,
+        opctx: &OpContext,
+        address: SocketAddr,
+    ) -> ListResultVec<Volume> {
+        opctx.check_complex_operations_allowed()?;
+
+        let mut volumes = Vec::new();
+        let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let needle = address.to_string();
+
+        while let Some(p) = paginator.next() {
+            use db::schema::volume::dsl;
+
+            let haystack =
+                paginated(dsl::volume, dsl::id, &p.current_pagparams())
+                    .select(Volume::as_select())
+                    .get_results_async::<Volume>(&*conn)
+                    .await
+                    .map_err(|e| {
+                        public_error_from_diesel(e, ErrorHandler::Server)
+                    })?;
+
+            paginator = p.found_batch(&haystack, &|r| r.id());
+
+            for volume in haystack {
+                if volume.data().contains(&needle) {
```

Comment:
Does this.. ah, it's just a string match somewhere in the JSON.

Reply:
YEP